src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -57,6 +57,15 @@ function getUnmaskedContext(workInProgress : Fiber) : Object {
 }
 exports.getUnmaskedContext = getUnmaskedContext;
 
+function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
+  const instance = workInProgress.stateNode;
+  if (instance) {
```

Comment:
Generally I'm of the opinion that methods should be strict and either have a safety check inside (and we never do it outside) or have no safety check, and we must always do it outside.

Doing it both ways means somebody will see the "early check" pattern that was made for performance reasons and scratch their head why other calls aren't behind the same check. This will look "surprising" if one call is different from other calls in the same file. And every difference like this is mental overhead when you have to move those pieces around. Strict contracts help avoid that, and we can always fix safety concerns with tests for regressions.